### PR TITLE
INT-5301 - Ingest AzureDevOps Repos

### PR DESCRIPTION
**Pull request by eXtremeX**
### Updated

- Updated SDK package versions to v8

### Added

- Added new entity `azure_devops_repo`
- Added new mapped relationship `azure_devops_project_uses_repo`
- Added new relationship `azure_devops_project_uses_repo` if repo isn't github
  or bitbucket

### CODE REVIEW

File: src/steps/users/index.ts [hunk 1]
```diff
@@ -40,12 +40,13 @@ export async function fetchUsers({
             name: user.identity?.displayName,
             displayName: user.identity?.displayName,
             email: user.identity?.uniqueName,
-            emailDomain: user.identity?.uniqueName?.split('@').pop(),
+            emailDomain: [user.identity?.uniqueName?.split('@').pop() || ''],
```
Comment (adam-in-ict) [inline src/steps/users/index.ts:43]: Do we mean to make the emailDomain an array?

File: docs/jupiterone.md [hunk 2]
```diff
@@ -96,10 +98,19 @@ The following relationships are created/mapped:
...
 | `azure_devops_user`    | **ASSIGNED**          | `azure_devops_work_item` |
 | `azure_devops_user`    | **CREATED**           | `azure_devops_work_item` |
 
+### Mapped Relationships
+
+The following mapped relationships are created:
+
+| Source Entity `_type`  | Relationship `_class` | Target Entity `_type` | Direction |
+| ---------------------- | --------------------- | --------------------- | --------- |
+| `azure_devops_project` | **USES**              | `*azure_devops_repo*` | FORWARD   |
```
Comment (adam-in-ict) [inline docs/jupiterone.md:112]: I think this is autogenerated, but it looks odd.  I think the target is based on a class not a _type for this mapped relationship.  Probably nothing to be done here, but might be worth noting for future updates to the SDK.

File: src/steps/repos/index.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        sourceType: 'azure_devops_project',
+        targetType: 'azure_devops_repo',
+      },
+    ],
+    mappedRelationships: [
+      {
+        _type: 'azure_devops_project_uses_repo',
+        _class: RelationshipClass.USES,
+        sourceType: 'azure_devops_project',
+        targetType: 'azure_devops_repo',
```
Comment (adam-in-ict) [inline src/steps/repos/index.ts:120]: is this the correct targetType for this mapped relationship?

**Comment by adam-in-ict** [comment]
A few high level things we need to do for this PR:
- Remove gitleaks from the .github/workflows folder as well as the .gitleaks.yml file from the root directory
- Upgrade the build.yml github workflow to target node-version 14.x.  You should be able to base this off of what's [currently in the integration-template project](https://github.com/JupiterOne/integration-template/blob/main/.github/workflows/build.yml).

If you keep getting build errors for 12.x, let me know and I can get an admin tracked down to clear it out.

File: src/steps/users/index.ts (hunk 1 above)
Comment (eXtremeX) [reply src/steps/users/index.ts:43]: Good question! It needs to be an array (https://github.com/JupiterOne/data-model/blob/main/src/schemas/User.json#L19), that's the only reason we've done it that way. Would you change that?

Comment (adam-in-ict) [reply src/steps/users/index.ts:43]: Nope!  I didn't realize that's what was in the data-model.  Thanks!

**Review by adam-in-ict — APPROVED** [review]

**Pull request merged**